packages/core/utils/lib/convert-query-params.js

```diff
@@ -139,6 +139,41 @@ const convertLimitQueryParams = (limitQuery) => {
   return limitAsANumber;
 };
 
+const convertPageQueryParams = (page) => {
+  const pageVal = toNumber(page);
+
+  if (!isInteger(pageVal) || pageVal <= 0) {
+    throw new PaginationError(
+      `Invalid 'page' parameter. Expected an integer > 0, received: ${page}`
+    );
+  }
+
+  return pageVal;
+};
+
+const convertPageSizeQueryParams = (pageSize, page) => {
+  const pageSizeVal = toNumber(pageSize);
+
+  if (!isInteger(pageSizeVal) || pageSizeVal <= 0) {
+    throw new PaginationError(
+      `Invalid 'pageSize' parameter. Expected an integer > 0, received: ${page}`
+    );
+  }
+
+  return pageSizeVal;
+};
+
+const validatePaginationParams = (page, pageSize, start, limit) => {
+  const isPagePagination = !isNil(page) || !isNil(pageSize);
+  const isOffsetPagination = !isNil(start) || !isNil(limit);
+
+  if (isPagePagination && isOffsetPagination) {
+    throw new PaginationError(
+      'Invalid pagination attributes. You cannot use page and offset pagination in the same query'
+    );
+  }
+};
```

Comment:
Shouldn't we use Yup validation?

Reply:
as this is in the entity service, which can be used outside our controllers, I see value in revalidaitng it here.

Also it is used in graphql now, so having it here makes it easy imo. wdyt?

Reply:
Sure, there is value to check data here as it can be used outside our controllers. But should we use Yup here too? 🤔

Reply:
do you mean using the yup validation in this layer, or in the api / graphl layers?

Reply:
I am not sure we need validations in the @strapi/core level, as there are not many we do (most are done in the upper layers).

Do you think it's worth adding the package in strapi core?  

What use cases would it fulfill here? I only see this validatePaginationParams. And I feel it would add more verbosity just to add a yup validation schema here.